### PR TITLE
Use bundled kotlin plugin instead kotlin-embeddable dependency

### PR DESCRIPTION
**Pull request by egorikftp**
Continue of https://github.com/ComposeGears/Valkyrie/pull/105

image

### CODE REVIEW

File: components/psi/build.gradle.kts [hunk 1]
```diff
@@ -1,15 +1,25 @@
...
-    // https://plugins.jetbrains.com/docs/intellij/using-kotlin.html#kotlin-standard-library
-    compileOnly(libs.kotlin.stdlib)
-
-    implementation(libs.kotlin.embeddable)
-
     testImplementation(libs.bundles.test)
     testRuntimeOnly(libs.junit.launcher)
+
+    intellijPlatform {
+        intellijIdeaCommunity("2024.1")
```
Comment (Goooler) [inline components/psi/build.gradle.kts:20]: This version could be extracted to version catalog.

File: components/psi/build.gradle.kts [hunk 2]
```diff
@@ -1,15 +1,25 @@
...
+        instrumentationTools()
+
+        bundledPlugin("org.jetbrains.kotlin")
```
Comment (Goooler) [inline components/psi/build.gradle.kts:23]: Looks like these dependencies can't be accessed from test sourceset.

Comment (egorikftp) [reply components/psi/build.gradle.kts:23]: the problem for now is how to use `com.intellij.openapi.project.Project` in tests 🙂 
will investigate

Comment (egorikftp) [reply components/psi/build.gradle.kts:27]: @Goooler could you please check when you have time?

File: playground/app/build.gradle.kts [hunk 3]
```diff
@@ -19,7 +19,7 @@ android {
 
     buildTypes {
         release {
-            isMinifyEnabled = true
+            isMinifyEnabled = false
```
Comment (Goooler) [inline playground/app/build.gradle.kts:22]: What's the story here?

File: components/psi/build.gradle.kts (hunk 2 above)
Comment (Goooler) [reply components/psi/build.gradle.kts]: A small tweak in f97742986283d03ab693987ef8bb4d5ade6bca0e.

**Review by Goooler — APPROVED** [review]
It works well!

File: playground/app/build.gradle.kts (hunk 3 above)
Comment (egorikftp) [reply playground/app/build.gradle.kts:22]: During the Gradle build command, it tries to minify the release APK, which seems like an unnecessary waste of time.

Comment (Goooler) [reply playground/app/build.gradle.kts:22]: yeah

File: idea-plugin/build.gradle.kts [hunk 4]
```diff
@@ -47,9 +46,13 @@ dependencies {
...
     // https://plugins.jetbrains.com/docs/intellij/tools-intellij-platform-gradle-plugin.html#setting-up-intellij-platform
     intellijPlatform {
-        intellijIdeaCommunity("2024.1")
+        intellijIdeaCommunity(libs.versions.idea)
         instrumentationTools()
         zipSigner()
+
+        // dependency plugin id for https://plugins.jetbrains.com/plugin/6954-kotlin
+        bundledPlugin(libs.kotlin.stdlib.map(Dependency::getGroup))
+        localPlugin(projects.components.psi)
```
Comment (egorikftp) [inline idea-plugin/build.gradle.kts:55]: @Goooler maybe this break release

https://github.com/ComposeGears/Valkyrie/actions/runs/10330536890

Package 'io.github.composegears.valkyrie.psi' is not found
          Package 'io.github.composegears.valkyrie.psi' is not found along with its 2 classes.
          Probably the package 'io.github.composegears.valkyrie.psi' belongs to a library or dependency that is not resolved by the checker.
          It is also possible, however, that this package was actually removed from a dependency causing the detected problems. Access to unresolved classes at runtime may lead to **NoSuchClassError**.
          The following classes of 'io.github.composegears.valkyrie.psi' are not resolved:
            Class io.github.composegears.valkyrie.psi.IconPackInfo is referenced in
              io.github.composegears.valkyrie.ui.screen.mode.iconpack.existingpack.ui.viewmodel.ExistingPackViewModel.onChooseFile(Path path, Project project) : void

Comment (Goooler) [reply idea-plugin/build.gradle.kts:55]: 56e2984353d55530f0cb3cac5410a3a79b9944ae

Wait https://github.com/ComposeGears/Valkyrie/actions/runs/10330644000/job/28599926138.

**Pull request merged**